x/auth/module.go

```diff
@@ -185,3 +191,33 @@ func (am AppModule) RegisterStoreDecoder(sdr sdk.StoreDecoderRegistry) {
 func (AppModule) WeightedOperations(_ module.SimulationState) []simtypes.WeightedOperation {
 	return nil
 }
+
+//
+// New App Wiring Setup
+//
+
+func init() {
+	appmodule.Register(&modulev1.Module{},
+		appmodule.Provide(provideModuleBasic, provideModule),
+	)
+}
+
+func provideModuleBasic() runtime.AppModuleBasicWrapper {
+	return runtime.WrapAppModuleBasic(AppModuleBasic{})
+}
+
+func provideModule(
+	config *modulev1.Module,
+	key *store.KVStoreKey,
+	cdc codec.Codec,
+	subspace paramtypes.Subspace) (keeper.AccountKeeper, runtime.AppModuleWrapper) {
+
+	maccPerms := map[string][]string{}
+	for _, permission := range config.ModuleAccountPermissions {
+		maccPerms[permission.Account] = permission.Permissions
+	}
+
+	k := keeper.NewAccountKeeper(cdc, key, subspace, types.ProtoBaseAccount, maccPerms, config.Bech32Prefix)
+	m := NewAppModule(cdc, k, simulation.RandomGenesisAccounts)
```

Comment:
`types.ProtoBaseAccount` and `simulation.RandomGenesisAccounts` are used here always instead of having configuration parameters. I propose that we handle this configuration in a separate PR. See #12055